plansys2_problem_expert/src/plansys2_problem_expert/ProblemExpert.cpp

```diff
@@ -122,7 +122,7 @@ ProblemExpert::addPredicate(const Predicate & predicate)
       return false;
     }
   } else {
-    return false;
+    return true; // nothing wrong with adding a predicate twice, but don't want to add it twice to the vector
```

Comment:
I think that the semantic are understandable without adding too verbose comments.